src/g_game.c

```diff
@@ -4067,6 +4068,20 @@ void doomprintf(int category, const char *s, ...)
   players[displayplayer].message = msg;  // set new message
 }
 
+void pickupmsg(player_t *player, const char *s, ...)
+{
+  static char msg[MAX_MESSAGE_SIZE];
+  va_list v;
+
+  if (player != &players[displayplayer] || !show_pickup_messages)
```

Comment:
From your testing, do we actually need this check? Since we're pointing to a specific player, wouldn't that assign the message to just that player?

Reply:
Right, we don't need `player != &players[displayplayer]` check.